File: 151-reverse-words-in-a-string/151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        
        s+=" ";
        string r = "";
        string e ="";
        for(auto c:s)
        {
            if(c==' ')
            {
                if(r.size()==0)
                    r = e;
                
                else if(e.size()!=0)
                    r = e+" "+r;
                
                e = "";
            }
            else
                e+=c;
        }
        return r;
    }
};

File: 151-reverse-words-in-a-string/151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        s+=" ";
        String u = "";
        String r = "";
        for(int i =0;i<s.length();++i)
        {
            if(s.charAt(i)==' ')
            {
                if(r.length()==0)
                    continue;
                if(u.length()==0)
                    u =r;
                else
                {
                    u = r+ " "+u;
                    
                }
                // System.out.println(r);
                
                r = "";
            }
            else
                r+=s.charAt(i);
        }
        return u;
    }
}